### ISSUE TITLE
Database size grows too quickly

### ISSUE THREAD

**Issue by 3noch** [agentm/project-m36#146]
My *very* rough benchmarks are showing that a very simple database schema can grow extremely fast.

A database with about 10 total attributes (between two relations) gets to a size of 16MB after 31 transactions (in which each transaction does exactly one insert). I think the size will go up exponentially.

**Comment by agentm** [comment]
I have been testing database growth with:
```
cabal run handles -- -d /tmp/handles -o 1 -t 100 -s "x:=relation{tuple{a 0}}" -i "insert x (x group({a} as b):{a:=count(@b)}){a}"
```
which creates a single-attribute relation with an ever-increasing integer on each commit.

**Comment by agentm** [comment]
By storing atom functions in a single file instead of in individual files and by using gzip compression on relvars, I managed to shave down 4-5x of the storage cost to a minimum of 48K per transaction for a single additional single-attributed relvar. That's still rather excessive, but a new on-disk format is in the works which should alleviate this. For now, I think the excessive growth has been curbed, but I recognize this is a half-measure.

Merged in d59c0f75a205e62b70997fe852d5eb246343ee6e.
